modules/account-lib/src/coin/stx/transactionBuilder.ts

```diff
@@ -0,0 +1,241 @@
+    // so we set the source here and leave the actual signing for the build step
+    this._multiSignerKeyPairs.push(signer);
+    this._numberSignatures = this._multiSignerKeyPairs.length > 1 ? this._multiSignerKeyPairs.length - 1 : this._multiSignerKeyPairs.length;
+    return this.transaction;
+  }
+
+  /** @inheritdoc */
+  protected get transaction(): Transaction {
+    return this._transaction;
+  }
+
+  /** @inheritdoc */
+  protected set transaction(transaction: Transaction) {
+    this._transaction = transaction;
+  }
+
+  /**
+   * Validates that the given key is not already in this._multiSignerKeyPairs
+   *
+   * @param {BaseKey} key - The key to check
+   */
+  private checkDuplicatedKeys(key: BaseKey) {
+    this._multiSignerKeyPairs.forEach(_sourceKeyPair => {
+      if (_sourceKeyPair.getKeys().prv === key.key) {
+        throw new SigningError('Repeated sign: ' + key.key);
+      }
+    });
+  }
+
+  /**
+   * Set the transaction fees
+   *
+   * @param {BaseFee} fee The maximum gas to pay
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  fee(fee: BaseFee): this {
+    this.validateValue(new BigNumber(fee.fee));
+    this._fee = fee;
+    return this;
+  }
+
+  network(stacksNetwork: StacksNetwork): this {
+    this._network = stacksNetwork;
+    return this;
+  }
+
+  nonce(n: number): this {
+    this._nonce = n;
+    return this;
+  }
+
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    this._memo = memo;
+    return this;
+  }
+
+  /**
+   *  Set the number of signatures for multi-sig
+   *
+   * @param {number} numSigns
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  numberSignatures(numSigns: number): this {
+    this.validateValue(new BigNumber(numSigns));
+    this._numberSignatures = numSigns;
+    return this;
+  }
+
+  // region Validators
+  /** @inheritdoc */
+  validateAddress(address: BaseAddress, addressFormat?: string): void {
+    if (!isValidAddress(address.address)) {
+      throw new BuildTransactionError('Invalid address ' + address.address);
+    }
+  }
+
+  /** @inheritdoc */
+  validateKey(key: BaseKey): void {
+    const keyPair = new KeyPair({ prv: key.key });
+    if (!keyPair.getKeys().prv) {
+      throw new BuildTransactionError('Invalid key');
+    }
+  }
+
+  /** @inheritdoc */
+  validateRawTransaction(rawTransaction: string): void {
+    if (!rawTransaction) {
+      throw new InvalidTransactionError('Raw transaction is empty');
+    }
+    try {
+      deserializeTransaction(BufferReader.fromBuffer(Buffer.from(removeHexPrefix(rawTransaction), 'hex')));
+    } catch (e) {
+      throw new ParseTransactionError('There was an error parsing the raw transaction');
+    }
+  }
+
+  /** @inheritdoc */
+  validateTransaction(transaction?: Transaction): void {
+    this.validateFee();
```

Comment:
are there no other checks in validateTransaction? Just validate fee?

Reply:
yes, validateNonce should be there. Adding it. Could not think of any other validations at this point.